<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>createEnvironment</class>
 <widget class="QWidget" name="createEnvironment">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1307</width>
    <height>721</height>
   </rect>
  </property>
  <property name="focusPolicy">
   <enum>Qt::NoFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Create Environment</string>
  </property>
  <widget class="createEnvironmentOpenglWindow" name="widget" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>850</width>
     <height>700</height>
    </rect>
   </property>
  </widget>
  <widget class="QListWidget" name="listWidget">
   <property name="geometry">
    <rect>
     <x>881</x>
     <y>180</y>
     <width>258</width>
     <height>71</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="deleteButton">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1020</x>
     <y>260</y>
     <width>120</width>
     <height>28</height>
    </rect>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
  </widget>
  <widget class="QPushButton" name="newEnvButton">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>10</y>
     <width>170</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>New Environment</string>
   </property>
  </widget>
  <widget class="QPushButton" name="loadButton">
   <property name="geometry">
    <rect>
     <x>1090</x>
     <y>10</y>
     <width>180</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Load Environment</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_1">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>50</y>
     <width>161</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Environment Name :</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>80</y>
     <width>581</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>..</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>150</y>
     <width>201</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Objects in Environment :</string>
   </property>
  </widget>
  <widget class="QPushButton" name="editButton">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>260</y>
     <width>120</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Edit</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>370</y>
     <width>70</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Size :</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>510</y>
     <width>71</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Location :</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="sizeXspinBox">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>920</x>
     <y>390</y>
     <width>64</width>
     <height>28</height>
    </rect>
   </property>
   <property name="maximum">
    <number>2700</number>
   </property>
   <property name="singleStep">
    <number>10</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>890</x>
     <y>400</y>
     <width>21</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>X :</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="sizeYspinBox">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1020</x>
     <y>390</y>
     <width>64</width>
     <height>28</height>
    </rect>
   </property>
   <property name="maximum">
    <number>2700</number>
   </property>
   <property name="singleStep">
    <number>10</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>990</x>
     <y>400</y>
     <width>21</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Y :</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="sizeZspinBox">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1120</x>
     <y>390</y>
     <width>64</width>
     <height>28</height>
    </rect>
   </property>
   <property name="maximum">
    <number>2700</number>
   </property>
   <property name="singleStep">
    <number>10</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>1090</x>
     <y>400</y>
     <width>21</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Z :</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>1090</x>
     <y>550</y>
     <width>21</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Z :</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="locationXspinBox">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>920</x>
     <y>540</y>
     <width>64</width>
     <height>28</height>
    </rect>
   </property>
   <property name="maximum">
    <number>2700</number>
   </property>
   <property name="singleStep">
    <number>10</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>890</x>
     <y>550</y>
     <width>21</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>X :</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>990</x>
     <y>550</y>
     <width>21</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string>Y :</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="locationZspinBox">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1120</x>
     <y>540</y>
     <width>64</width>
     <height>28</height>
    </rect>
   </property>
   <property name="maximum">
    <number>2700</number>
   </property>
   <property name="singleStep">
    <number>10</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="locationYspinBox">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1020</x>
     <y>540</y>
     <width>64</width>
     <height>28</height>
    </rect>
   </property>
   <property name="maximum">
    <number>2700</number>
   </property>
   <property name="singleStep">
    <number>10</number>
   </property>
  </widget>
  <widget class="QPushButton" name="addButton">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>300</y>
     <width>260</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Add Cuboid</string>
   </property>
  </widget>
  <widget class="QPushButton" name="sizeUndoButton">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>920</x>
     <y>440</y>
     <width>60</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
  </widget>
  <widget class="QPushButton" name="sizeResetButton">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1000</x>
     <y>440</y>
     <width>60</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Reset</string>
   </property>
  </widget>
  <widget class="QPushButton" name="locationUndoButton">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>920</x>
     <y>590</y>
     <width>60</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
  </widget>
  <widget class="QPushButton" name="locationResetButton">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1000</x>
     <y>590</y>
     <width>60</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Reset</string>
   </property>
  </widget>
  <widget class="QPushButton" name="saveDimentionButton">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1210</x>
     <y>360</y>
     <width>61</width>
     <height>260</height>
    </rect>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>340</y>
     <width>650</width>
     <height>18</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="saveButton">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>920</x>
     <y>670</y>
     <width>104</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </widget>
  <widget class="QPushButton" name="cancelButton">
   <property name="geometry">
    <rect>
     <x>1040</x>
     <y>670</y>
     <width>104</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Save as</string>
   </property>
  </widget>
  <widget class="QPushButton" name="exitButton">
   <property name="geometry">
    <rect>
     <x>1160</x>
     <y>670</y>
     <width>104</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>120</y>
     <width>141</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Floor Image :</string>
   </property>
  </widget>
  <widget class="QPushButton" name="browseButton">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1180</x>
     <y>112</y>
     <width>90</width>
     <height>30</height>
    </rect>
   </property>
   <property name="text">
    <string>Browse</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEdit">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>972</x>
     <y>113</y>
     <width>191</width>
     <height>27</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>createEnvironmentOpenglWindow</class>
   <extends>QWidget</extends>
   <header>createenvironmentopenglwindow.h</header>
   <container>1</container>
   <slots>
    <slot>testSlot()</slot>
    <slot>updateObject(vector_envObjectType)</slot>
    <slot>currentObject(QString)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>listWidget</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>createEnvironment</receiver>
   <slot>currentRowChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>881</x>
     <y>240</y>
    </hint>
    <hint type="destinationlabel">
     <x>545</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newEnvButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>newEnvButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>907</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>918</x>
     <y>25</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loadButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>loadEnvButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1115</x>
     <y>27</y>
    </hint>
    <hint type="destinationlabel">
     <x>1129</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>editObjectButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>999</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>1091</x>
     <y>129</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>deleteObjectButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1139</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>1195</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>addObjectButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1139</x>
     <y>320</y>
    </hint>
    <hint type="destinationlabel">
     <x>1233</x>
     <y>191</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listWidget</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>label_13</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>937</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>954</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sizeUndoButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>sizeUndoButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1083</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>1069</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sizeResetButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>sizeResetButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1152</x>
     <y>301</y>
    </hint>
    <hint type="destinationlabel">
     <x>1137</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationUndoButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>locationUndoButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1083</x>
     <y>367</y>
    </hint>
    <hint type="destinationlabel">
     <x>1035</x>
     <y>347</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationResetButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>locationResetButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1149</x>
     <y>369</y>
    </hint>
    <hint type="destinationlabel">
     <x>1115</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveDimentionButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>saveDimentionButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1240</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>1208</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>createEnvironment</sender>
   <signal>objectUpdated(vector_envObjectType)</signal>
   <receiver>widget</receiver>
   <slot>updateObject(vector_envObjectType)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>925</x>
     <y>590</y>
    </hint>
    <hint type="destinationlabel">
     <x>662</x>
     <y>579</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>saveButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>956</x>
     <y>674</y>
    </hint>
    <hint type="destinationlabel">
     <x>1070</x>
     <y>658</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>cancelButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1107</x>
     <y>682</y>
    </hint>
    <hint type="destinationlabel">
     <x>1209</x>
     <y>638</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sizeXspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>createEnvironment</receiver>
   <slot>sizeXspinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>801</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>731</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sizeYspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>createEnvironment</receiver>
   <slot>sizeYspinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>890</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>732</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sizeZspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>createEnvironment</receiver>
   <slot>sizeZspinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>991</x>
     <y>301</y>
    </hint>
    <hint type="destinationlabel">
     <x>728</x>
     <y>315</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationXspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>createEnvironment</receiver>
   <slot>locationXspinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>782</x>
     <y>366</y>
    </hint>
    <hint type="destinationlabel">
     <x>755</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationYspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>createEnvironment</receiver>
   <slot>locationYspinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>881</x>
     <y>384</y>
    </hint>
    <hint type="destinationlabel">
     <x>756</x>
     <y>426</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationZspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>createEnvironment</receiver>
   <slot>locationZspinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>971</x>
     <y>382</y>
    </hint>
    <hint type="destinationlabel">
     <x>755</x>
     <y>451</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listWidget</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>widget</receiver>
   <slot>currentObject(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>881</x>
     <y>211</y>
    </hint>
    <hint type="destinationlabel">
     <x>694</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exitButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>exitButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1239</x>
     <y>681</y>
    </hint>
    <hint type="destinationlabel">
     <x>1263</x>
     <y>656</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>browseButton</sender>
   <signal>clicked()</signal>
   <receiver>createEnvironment</receiver>
   <slot>browseButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1208</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>1208</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>testSignal()</signal>
  <signal>objectUpdated(vector_envObjectType)</signal>
  <slot>del()</slot>
  <slot>currentRowChanged(int)</slot>
  <slot>createNewEnvironment()</slot>
  <slot>loadEnvironment()</slot>
  <slot>newEnvButtonClicked()</slot>
  <slot>loadEnvButtonClicked()</slot>
  <slot>addObjectButtonClicked()</slot>
  <slot>deleteObjectButtonClicked()</slot>
  <slot>editObjectButtonClicked()</slot>
  <slot>sizeUndoButtonClicked()</slot>
  <slot>sizeResetButtonClicked()</slot>
  <slot>locationUndoButtonClicked()</slot>
  <slot>locationResetButtonClicked()</slot>
  <slot>saveDimentionButtonClicked()</slot>
  <slot>saveButtonClicked()</slot>
  <slot>cancelButtonClicked()</slot>
  <slot>sizeXspinBoxValueChanged(int)</slot>
  <slot>sizeYspinBoxValueChanged(int)</slot>
  <slot>sizeZspinBoxValueChanged(int)</slot>
  <slot>locationXspinBoxValueChanged(int)</slot>
  <slot>locationYspinBoxValueChanged(int)</slot>
  <slot>locationZspinBoxValueChanged(int)</slot>
  <slot>exitButtonClicked()</slot>
  <slot>browseButtonClicked()</slot>
 </slots>
</ui>
